lib/elixir/lib/kernel.ex

```diff
@@ -2303,6 +2308,10 @@ defmodule Kernel do
     end)
   end
 
+  defp build_if(_condition, _keys) do
+    raise(ArgumentError, "invalid or duplicate keys for if")
```

Comment:
Change keys to arguments or maybe blocks. Maybe we can also print the keys that are invalid.

Same for build_unless.